lib/engine/game/g_1832/step/token.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/token'
+
+module Engine
+  module Game
+    module G1832
+      module Step
+        class Token < Engine::Step::Token
+          def actions(entity)
+            return super unless can_place_miami_token?(entity)
+
+            super | %w[choose pass]
+          end
+
+          def choice_name
+            'Additional Token Actions'
+          end
+
+          def choices
+            choices = {}
+            choices['keywest'] = 'Place Key West Token' if can_place_miami_token?(current_entity)
+
+            choices
+          end
+
+          def process_choose(action)
+            raise GameError, 'Illegal choice' unless action.choice == 'keywest'
+
+            place_miami_token(action.entity)
+            pass!
+          end
+
+          def place_miami_token(corporation)
+            ability = @game.abilities(corporation, :assign_hexes)
+            hex = @game.hex_by_id(ability.hexes.first)
+
+            hex.tile.icons.reject! { |icon| icon.name == 'FECR_key_west' }
+            hex.assign!(corporation)
```

Comment:
I'd missed this one when pointing out the problems with `can_place_miami_token?` – this will also break if a corporation has two abilities of type `assign_hexes`. You need to do something similar here, using `Array()` to make sure that you get a list of abilities back from `@game.abilities`, then finding the Miami ability in those.